pkg/actions/flux/enable.go

```diff
@@ -45,6 +57,35 @@ func New(k8sClientSet kubeclient.Interface, opts *api.GitOps) (*Installer, error
 	return installer, nil
 }
 
+func (ti *Installer) InjectIRSA(irsas *[]outputs.IRSA) error {
+	if len(*irsas) != 0 {
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fluxNS,
+			},
+		}
+		if _, err := ti.kubeClient.CoreV1().Namespaces().Create(context.TODO(), ns, metav1.CreateOptions{}); err != nil {
```

Comment:
This has to be an ensure in case the namespace happens to already exist. There are examples of ensure elsewhere in the code base.

Reply:
Also the other create needs to be ab ensure too.

Reply:
This code is called as a result of eksctl create cluster -- so there is no flux-system namespace or anything else in it. The bootstrap happens after this inject. I have tested this extensively and all works fine.

Reply:
It might work in isolation but on a cluster where other things might be running or in a multi-tenant environment, or just when running in parallel with other things, this might be caught in a situation where it collides with another process doing a similar thing. :)

You're right that the probability is low, but it's never zero. :) So we have things like ensure for that situation.

Reply:
Appreciate if you can point me to an example of ensure usage like I would need. Thanks

Reply:
I'll check it, it isn't called ensure, it's a Kubernetes API thing. Ahh I can't remember what it's called. :D Something like MaybeCreate or Maybe... something, something.

Reply:
Hah, found it. :) MaybeCreateNamespace -> look for this. There is a file called namespace.go. :)

Reply:
Yes, @Skarlso will look at that. Thanks